articles/121_realtime_proposal.md

```diff
@@ -0,0 +1,230 @@
+---
+layout: default
+title: Proposal for Implementation of Real-time Systems in ROS 2
+permalink: articles/realtime_proposal.html
+abstract: This article proposes an approach to the real-time performance requirement in ROS 2.
+published: true 
+author: Jackie Kay
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Requirements and Implementation of Real-Time Systems
+
+### System requirements
+
+The requirements of a real-time system varies depending on the use case.
+At its core, the real-time requirement of a system has two components:
+
+- Latency
+
+  - Update period
+  - Determinism (measured by jitter)
+
+- Failure mode
+
+  - How to react to a missed deadline
+
+Language referring to hard/soft/firm real-time systems generally refers to the failure mode.
+A hard real-time system treats a missed deadline as a system failure.
+A soft real-time system tries to meet deadlines but does not fail when a deadline is missed.
+A firm real-time system discards computations made by missed deadlines and may degrade its performance requirement in order to adapt to a missed deadline.
+
+Though in this document we consider the failure mode orthogonal to latency characteristics, hard real-time systems are usually designed to be safety critical and therefore require high determinism (low jitter).
```

Comment:
`hard real-time systems are usually designed to be safety critical`

I would say it is the other way around:

`safety critical systems are usually designed to be hard real-time`?